swig/lua52/CMakeLists.txt

```diff
@@ -0,0 +1,45 @@
+add_definitions(-DSWIG)
+add_definitions(-DSWIGLUA)
+
+project(Sysrepo_Lua)
+
+add_library(Sysrepo_Lua SHARED ${SYSREPO_CPP_SOURCES})
+
+# include custom Modules
+set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}/CMakeModules/")
+
+include_directories(${CMAKE_SOURCE_DIR})
+include_directories(${CMAKE_CURRENT_SOURCE_DIR}/src)
+target_link_libraries(Sysrepo_Lua sysrepo)
+
+if(NOT BIN_INSTALL_DIR)
+	set(BIN_INSTALL_DIR ${CMAKE_INSTALL_PREFIX}/bin)
+endif()
+
+set(LUA_SWIG_BINDING_52 libsysrepoLua52)
+
+if (SWIG_FOUND)
+    include(${SWIG_USE_FILE})
+    find_package(Lua 5.2 REQUIRED)
```

Comment:
> I do not want that the lua5.2 bindings are built with the lua5.1 library

I agree, but changing `find_package(Lua 5.2 REQUIRED)` to `find_package(Lua 5.2)` would not allow it, would it? You can also check `LUA_VERSION_STRING` or `LUA_VERSION_MAJOR` and `LUA_VERSION_MINOR` in the branch below.

> I was actually thinking of adding a check in main cmake file. Something like:
lua:
selected version:
supported versions:
available versions:
And if there is a mismatch I would change the selected version and display a message.
I could do the same for python.

Yes, that is a good idea. In any case - if there is anything missing (python2/python3/lua51/lua52), the build of sysrepo must always succeed - we should just just print a warning.